MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -721,29 +724,60 @@ public final class MagazineLayout: UICollectionViewLayout {
-    // example), we need to compensate for preferred size changes to items as we're scrolling up,
-    // otherwise, the collection view will appear to jump each time an element is sized.
-    // Since size adjustments can occur for multiple items in the same soon-to-be-visible row, we
-    // need to account for this by considering the preferred height for previously sized elements in
-    // the same row so that we only adjust the content offset by the exact amount needed to create
-    // smooth scrolling.
     let isScrolling = currentCollectionView.isDragging || currentCollectionView.isDecelerating
-    let isSizingElementAboveTopEdge = originalAttributes.frame.minY < currentCollectionView.contentOffset.y
-
-    if isScrolling && isSizingElementAboveTopEdge {
+    let top = currentCollectionView.bounds.minY + contentInset.top
+    let bottom = currentCollectionView.bounds.maxY
+    let isSizingElementAboveTopEdge = originalAttributes.frame.minY < top
+    let isSizingElementBelowBottomEdge = originalAttributes.frame.minY > bottom
+
+    let contentOffsetAdjustment: CGPoint
+    if verticalLayoutDirection == .topToBottom, isSizingElementAboveTopEdge {
+      // If layout information is discarded above our current scroll position (on rotation, for
+      // example), we need to compensate for preferred size changes to items as we're scrolling up,
+      // otherwise, the collection view will appear to jump each time an element is sized.
+      // Since size adjustments can occur for multiple items in the same soon-to-be-visible row, we
+      // need to account for this by considering the preferred height for previously sized elements
+      // in the same row so that we only adjust the content offset by the exact amount needed to
+      // create smooth scrolling.
       let isSameRowAsLastSizedElement = lastSizedElementMinY?.isEqual(
         to: currentElementY,
         threshold: 1 / scale)
         ?? false
+
       if isSameRowAsLastSizedElement {
         let lastSizedElementPreferredHeight = self.lastSizedElementPreferredHeight ?? 0
         if preferredAttributes.size.height > lastSizedElementPreferredHeight {
-          context.contentOffsetAdjustment.y = preferredAttributes.size.height - lastSizedElementPreferredHeight
+          contentOffsetAdjustment = CGPoint(
+            x: 0,
+            y: preferredAttributes.size.height - lastSizedElementPreferredHeight)
+        } else {
+          contentOffsetAdjustment = .zero
         }
       } else {
-        context.contentOffsetAdjustment.y = preferredAttributes.size.height - originalAttributes.size.height
+        contentOffsetAdjustment = CGPoint(
+          x: 0,
+          y: preferredAttributes.size.height - originalAttributes.size.height)
       }
+    } else if
+      verticalLayoutDirection == .bottomToTop,
+      !isSizingElementBelowBottomEdge,
+      modelState.isPerformingBatchUpdates
```

Comment:
I'm seeing in testing that we need this content offset even when batch updates aren't being performed. Here's a video of the below scenario where removing this check prevented a content offset jump. It's when you insert a cell at the top and it's offscreen.

https://github.com/airbnb/MagazineLayout/assets/1455522/7882de72-152a-4685-b747-a8184a4b4042